packages/@aws-cdk/cloud-assembly-schema/lib/schema.ts

```diff
@@ -0,0 +1,90 @@
+import * as fs from 'fs';
+import * as jsonschema from 'jsonschema';
+import * as assembly from './manifest';
+
+/**
+ * Protocol utility class.
+ */
+export class Manifest {
+  /**
+   * Save manifest to file.
+   *
+   * @param manifest - manifest.
+   */
+  public static save(manifest: assembly.AssemblyManifest, filePath: string) {
+    fs.writeFileSync(filePath, JSON.stringify(manifest, undefined, 2));
+  }
+
+  /**
+   * Load manifest from file.
+   */
+  public static load(filePath: string): assembly.AssemblyManifest {
+    const raw: assembly.AssemblyManifest = JSON.parse(fs.readFileSync(filePath, 'UTF-8'));
+    Manifest.patchStackTags(raw);
```

Comment:
I forget why this can't simply be done in the CLI? (i.e. the CLI should simply accept both capitalizations).

Reply:
You mean for when we start writing the tags with the lowercase? Actually to support this we need to change the `patch` method a bit, something like:

```typescript
metadataAny.data = metadataAny.data.map((t: any) => {
  // support both capitalizations 
  return { key: t.Key || t.key, value: t.Value || t.value };
});
```

Which we can do i guess, but we can also do this when we need to.

Did I misunderstand you?

Regardless, I feel this method belongs here because the CLI shouldn't really be aware of how the manifest is stored on disk, it should just work with the interfaces. Also, if you move it the CLI, consumers of this class wont be able to do:

```typescript
Manifest.load(manifest)...metadata[STACK_TAGS).data[0].key
```

Because the lowercase `key` property will be empty.